### DISCUSSION TITLE
Bug: When I save a session on Firefox, tabs' order gets mixed up badly

### DISCUSSION THREAD

**Discussion by jaimebibiloni** [sienori/Tab-Session-Manager#903] [Q&A]
The session gets saved normally but the issue arises when I open the saved session. The tabs are all over the place. **This has been an issue for months** but it has gotten worse recently. In the beginning it was only one or two tabs that were in different orders, but now the order of every tab seems just random.

However, **the bug does not happen if I save and then open the session on Chrome**, but it happens if I save on Chrome and open it on Firefox.

**The session I have had issues with has 28 tabs**.

**I have Firefox 93.0 (64-bit) installed, supposedly up to date**.

**Comment by loolllool** [comment]
I have the similar problem (FF #88): a bulk of tabs (>100) becomes a mess with every session restoring.
It seems they upload as chunks: one chunk of tabs by another, — with the own arrangement in each one.

**For example:** I had the active tab in closed session. 
The next session will start with that active tab and some tabs which were on the left and right side (~7).
Than it starts to upload other tabs after the rightmost (so, breaking the tabs' order) and do it from the oldest ones to recent. etc

**Reply by AENwww** [reply]
Yes, I find this to be quite a (1st-World) problem ...
Perhaps "we" might see a solution ... sooner ... or later ???
Hope so ...

**Comment by doktorsteb** [comment]
Hello, I have the same problem with pinned tabs. After each restart of the browser, the pinned tabs are in a random order. This problem has existed for years, but is very annoying. Is there a solution to this? I use Firefox 110.0 64 bit / Tab Session Manager v 6.12.1

**Reply by doktorsteb** [reply]
It seams the problem is since years unsolved. It's completely nerving that after some sessions the pinned tabs are going mixed up. So I give up. not the best Add-On by the way.
